src/api/app/components/rpm_lint_component.rb

```diff
@@ -0,0 +1,24 @@
+class RpmLintComponent < ApplicationComponent
+  attr_reader :badness_chart_data, :issues_chart_data
+
+  def render?
+    @errors.map(&:second).any?(&:positive?)
+  end
+
+  def initialize(rpmlint_log_parser:)
+    super
+
+    threshold_filter = ->(key, _value) { rpmlint_log_parser.errors[key].positive? }
+    badness_sorter = ->(a, b) { rpmlint_log_parser.badness[a[0]] <=> rpmlint_log_parser.badness[b[0]] }
+
+    @errors = rpmlint_log_parser.errors.select(&threshold_filter).sort(&badness_sorter).reverse
+    warnings = rpmlint_log_parser.warnings.select(&threshold_filter)
+    info = rpmlint_log_parser.info.select(&threshold_filter)
+    badness = rpmlint_log_parser.badness.select(&threshold_filter).sort(&badness_sorter).reverse
+
+    @issues_chart_data = [{ name: 'Errors', data: @errors },
+                          { name: 'Warnings', data: warnings },
+                          { name: 'Info', data: info }]
+    @badness_chart_data = [{ name: 'Badness', data: badness }]
+  end
+end
```

Comment:
Are we planning to cover specs for this component?

Reply:
Let me add it

Reply:
Done

Reply:
Thank you